src/app/compose/mailrecipientinput.component.ts

```diff
@@ -53,18 +54,24 @@ export class MailRecipientInputComponent implements OnInit, AfterViewInit {
     @ViewChild('searchTextInput') searchTextInput: ElementRef;
     @ViewChild('auto') auto: MatAutocomplete;
 
-    constructor(recipientservice: RecipientsService) {
+    constructor(
+        private snackBar: MatSnackBar,
+        recipientservice: RecipientsService
+    ) {
         recipientservice.recipients.subscribe((recipients) => {
 
         // Listen to search text input and popup suggestions from recipient list
         this.searchTextFormControl.valueChanges
             .pipe(debounceTime(50))
             .subscribe((searchtext: string) => {
                 if (searchtext) {
-                    const lowercaseSearchText = searchtext.toLowerCase();
+                    // I know that searchtext is supposed to be a string,
+                    // but somehow it ends up being the recipient object
```

Comment:
Would we want to figure out the "mystery" here?

Reply:
I'd love to, but I can't make sense of it: it keeps being text (as it should be), but somehow, the moment you click on it, it becomes the recipient object for a single tick, and for that single tick the searchformcontrol puts it through here.

Reply:
could it be because you converted the search index suggestions to recipient objects? In that case I guess it's ok.

Reply:
Yeah, that's probably that. Weird how it works alright in all other cases except right when you click them though :)

Reply:
The search text field will be a string, but the auto-complete popup results are Recipient objects. So maybe the appropriate would be to change line 66 to something like:

`.subscribe((searchtext: string | Recipient) => {`

Reply:
Ah, that's sensible. Will do :)